### PR TITLE
prov/verbs: Remove references to old FI_LOCAL_MR mode bit and add new v1.5 caps

### PR DESCRIPTION
**Pull request by a-ilango**
FI_MR_LOCAL domain mr_mode bit replaces FI_LOCAL_MR from v1.5 onwards. Remove any
references to FI_LOCAL_MR and use the domain mr_mode bit to infer MR registration
requirement.

Also add verbs/RDM related rx attr caps directly in the "prov_info". This removes
a check in fi_getinfo path. When app doesn't requrest FI_RMA, we avoid requiring
FI_RX_CQ_DATA.

### CODE REVIEW

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -260,8 +259,8 @@ int fi_ibv_check_rx_attr(const struct fi_rx_attr *attr,
 
 	compare_mode = attr->mode ? attr->mode : hints->mode;
 
-	check_mode = FI_IBV_EP_TYPE_IS_RDM(info) ? VERBS_RDM_MODE :
-		(hints->caps & FI_RMA) ? info->rx_attr->mode : VERBS_MODE;
+	check_mode = (hints->caps & FI_RMA) ? info->rx_attr->mode :
+		(info->rx_attr->mode & ~FI_RX_CQ_DATA);
```
Comment (gladkovdmitry17) [inline prov/verbs/src/verbs_info.c]: What do you think about `FI_RX_CQ_DATA`->`VERBS_RX_MODE`  ?

Comment (a-ilango) [inline prov/verbs/src/verbs_info.c]: This line disables requiring FI_RX_CQ_DATA mode if the app doesn't use FI_RMA. VERBS_RX_MODE could potentially include any other mode that doesn't concern FI_RMA. Although we don't have such a thing now.

Comment (gladkovdmitry17) [inline prov/verbs/src/verbs_info.c]: Thanks for the explanation

**Comment by a-ilango** [comment]
Thanks for the review!

**Pull request merged**